docs: polish and simplify README.md

adam3q opened the pull request:
1. Add FAQ, Website and Quickstart link.
2. Reorganize document structure.
3. Remove some redundant descriptions.

imeoer commented on README.md:
```diff
@@ -80,88 +85,79 @@ make release
-## Build Nydus Image
-
-Build Nydus image from directory source: [Nydus Image Builder](./docs/nydus-image.md).
+Build Nydus image from directory source: [Buildkit](https://github.com/moby/buildkit/blob/master/docs/nydus.md), [Nydus Image Builder](./docs/nydus-image.md).
```
How about

```
Convert OCI v1 image to nydus image: Nydusify, Acceld, Nerdctl.
Build nydus image from Dockerfile directly: Buildkit.
Build nydus layer from various sources: Nydus Builder.
```

adamqqqplay replied:
Fixed

adam3q committed:
docs: polish and simplify README.md

1. Add FAQ, Website and Quickstart link.
2. Reorganize document structure.
3. Remove some redundant descriptions.

Signed-off-by: Qinqi Qu <quqinqi@linux.alibaba.com>

imeoer reviewed (approved):
Thanks! LGTM.
